usImage.cpp

```diff
@@ -37,6 +37,68 @@
 
 #include <algorithm>
 
+class HistogramBuilder {
+    public:
+        int *histo;
+        unsigned short MinADU, MaxADU;
+        int pixCount;
+
+        HistogramBuilder() {
+            histo = new int[65536];
+            MinADU = 0;
+            MaxADU = 0;
+            pixCount = 0;
+        }
+
+        ~HistogramBuilder() {
+            delete histo;
+        }
+
+        unsigned short median() const
+        {
+            int pixelLeft = pixCount / 2;
+
+            for(int i = MinADU; i < MaxADU; ++i) {
+                if (histo[i] > pixelLeft) {
+                    return i;
+                }
+                pixelLeft -= histo[i];
+            }
+            return MaxADU;
+        }
+
+
+        void scan(unsigned short *t, int len)

```

Comment:
minor: we can use `const` to express (and enforce) that scan has no side-effects on `t`:
```suggestion
        void scan(const unsigned short *t, int len)
```